### PR TITLE
Add release-fedora to build fedora compatible RPM

### PR DESCRIPTION
**Pull request by filfreire**
## [SEE UPDATE 23 January 2023](https://github.com/Kong/insomnia/issues/5088#issuecomment-1400859407)
_____ 
In this PR we add another job ~~to the release-publish~~, `release-fedora` where we build Insomnia `.rpm` with `buildDependenciesFromSource` setting from within a fedora 36 container. 

This helps provide a working workaround build that solves the issue folks are running into for #5088.

When we do builds with `buildDependenciesFromSource` set to `true` from within a Fedora 36 environment, we no longer get the SSL crash:

Screenshot 2022-10-14 at 11 46 03

TL;DR: Exploration to try and provide some sort of workaround for #5088 

 > ⚠️ This should probably not be solved in the insomnia build, what we’re doing is making it build node-libcurl from source which means it will be different from other platforms and not statically compiled. So it’ll have other potential issues depending on the environment it’s ran on, and not necessarily behave the same as other platforms. The proper solution is probably to change the build of the prebuilt node-libcurl binaries so they’re compatible instead.

### CODE REVIEW

File: .github/workflows/release-publish.yml [hunk 1]
```diff
@@ -197,3 +197,32 @@ jobs:
...
+          docker create --name builder fedora-insomnia-builder:latest
+          docker cp builder:/insomnia/packages/insomnia/dist ./artifacts/
+          docker rm -f builder
+
+      - name: Upload artifacts to release
+        uses: xresloader/upload-to-github-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          release_id: ${{ steps.core_tag_and_release.outputs.id }}
```
Comment (filfreire) [inline .github/workflows/release-publish.yml]: not sure if `steps.core_tag_and_release.outputs.id` picks up the proper step from the other job

File: .github/workflows/release-publish.yml [hunk 2]
```diff
@@ -197,3 +197,32 @@ jobs:
...
+        with:
+          ref: ${{ env.RELEASE_BRANCH }}
+          fetch-depth: 0
+          persist-credentials: false
+
+      - name: Build Insomnia from sources for Fedora
+        run: |
+          docker build --tag fedora-insomnia-builder:latest -f fedora.Dockerfile .
+          docker create --name builder fedora-insomnia-builder:latest
+          docker cp builder:/insomnia/packages/insomnia/dist ./artifacts/
```
Comment (filfreire) [inline .github/workflows/release-publish.yml]: full path is copied into artifacts folder, needs to be adapted bellow

File: .github/workflows/release-publish.yml (hunk 1 above)
Comment (filfreire) [reply .github/workflows/release-publish.yml]: fixed

**Comment by azzamsa** [comment]
Is this fixes will land on master? 
I can't upgrade other packages because of pinned version of cups.

Thanks for insomnia! :heart:

**Comment by leonardehrenfried** [comment]
By the way, I tried the artifact from here: https://github.com/Kong/insomnia/actions/runs/3481857586

I still see the SSL connect error followed by a crash.

**Comment by filfreire** [comment]
Hi @azzamsa @leonardehrenfried , while we don't have a more definitive solution, in order to unblock you in the meantime, I've compiled an experimental build on my fork which you can download and use:

https://github.com/filfreire/insomnia/releases/tag/fedora-experiment%408ae4fe71

Contains changes up to [8ae4fe71](https://github.com/Kong/insomnia/commit/8ae4fe710dc7d80fcd14265e315f5c2f179821ab) and was built using: https://github.com/Kong/insomnia/pull/5281

Please let me know if you have any issues with it.

**Comment by leonardehrenfried** [comment]
@filfreire I just upgraded all of my dependencies and re-installed the build you provided. I can now indeed use insomnia again! Thank you for this.

This will be good enough for me for a while.

**Comment by filfreire** [comment]
## [SEE UPDATE 23 January 2023](https://github.com/Kong/insomnia/issues/5088#issuecomment-1400859407)
